### ISSUE TITLE
[a11y]: Date picker calendar not usable with JAWS

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#11438]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

https://react.carbondesignsystem.com/?path=/story/components-datepicker--single-with-calendar

### React version

https://react.carbondesignsystem.com/?path=/story/components-datepicker--single-with-calendar

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

The date picker is relatively keyboard accessible without JAWS running. (There are some limitations, like the inability to move by month via either a shortcut key or via the left < and right > arrows.)
The navigation is primary through the arrow key.
However, with JAWS running, this arrow key is disrupted by the screen reader.
It looks like it can be quite easily fixed by changing the "region" role on the parent container of the calendar to a role of "application". This allows JAWS to pass through the arrow keys to the calendar. At that point, JAWS correctly reads updates to the calendar position, including announcing when the month changes. It also properly announces when a date is chosen.

`<div class="flatpickr-calendar animate undefined--date-picker__calendar cds--date-picker__calendar open arrowTop arrowLeft" tabindex="-1" role="application" aria-label="calendar-container" style="top: 108px; left: 42px; right: auto;">`

### WCAG 2.1 Violation

2.1.1 Keyboard and Conformance requirement 5 

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?id=components-datepicker--single-with-calendar&args=&viewMode=story

### Steps to reproduce

Go to https://react.carbondesignsystem.com/iframe.html?id=components-datepicker--single-with-calendar&args=&viewMode=story

Without JAWS running
1. press Tab to enter date input (calendar appears)
2. press Down Arrow (focus moves to current date in calendar)
3. press Right Arrow 5 times (focus moves across calendar by day, wrapping to next week)
4. press Up Arrow 5 times (focus moves up by week and progresses to prior month, with a slightly odd glitch where it shifts to a different day as the month breaks oddly across weeks)

With JAWS running, repeat the above steps
**Expected behaviour is:**
1 JAWS enters forms mode (a beep will sound) and announces "Date Picker Label edit mm slash dd slash yyyy type and text"
2 JAWS reads May 18, 2022 Current Date
3 JAWS reads the dates sequentially
4. JAWS reads the dates that take focus, announces a new month when the calendar reverts to the prior month

**Actual behaviour is**
At step 2 JAWS drops out of forms mode and enters Virtual PC mode, so that pressing arrow keys moves between different parts of the calendar widget not expected to be navigable.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by dakahn** [comment]
@mbgower im going to turn that last comment into a separate issue 👍🏾
